libass/ass.h

```diff
@@ -258,6 +258,10 @@ void ass_library_done(ASS_Library *priv);
  * Optional directory that will be scanned for fonts recursively.  The fonts
  * found are used for font lookup.
  * NOTE: A valid font directory is not needed to support embedded fonts.
+ * On Microsoft Windows, when using WIN32-APIs, fonts_dir must either be
+ * in UTF-8 with possible unpaired surrogates and CESU-8 or the encoding accepted
+ * by fopen with UTF-8 taking precedence if both versions are valid and exist.
+ * On all other systems there is no need for special considerations like that.
```

Comment:
Nit: Instead of `with UTF-8 taking precedence` just `with the former taking precedence` would be more accurate as it's not really UTF-8.

To my understanding CESU-8 technically requires paired surrogates, which are individually encoded like UTF-8 as if they were valid codepoints and UTF-8 doesn't encompass the (UTF-16) surrogates. If that's correct `UTF-8 with possible unpaired surrogates and CESU-8` may be misleading. Perhaps something like this would be clearer?:
```
On Microsoft Windows, when using WIN32-APIs, fonts_dir must be in either
UTF-8 mixed with lone or paired UTF-16 surrogates encoded like in CESU-8
or the encoding accepted by fopen with the former taking precedence
if both versions are valid and exist.
On all other systems there is no need for special considerations like that.
```
Not sure if “lone or paired” is beneficial or just redundant.

Reply:
Just confirming: yes, my understanding is the same. UTF-8 requires 4-byte encoding for supplementary planes and forbids surrogate code points from being encoded at all. CESU-8 requires supplementary planes to be encoded as valid pairs of 3-byte surrogates and still forbids lone surrogates. We accept a mix of both _and_ lone surrogates.

Reply:
Adopted suggested text sample.